include/drivers/clock.h

```diff
@@ -18,6 +20,11 @@ struct clock_driver_ops {
 	int (*set_rate)(struct device *dev, uint8_t id, uint32_t rate);
 };
 
+struct clock_driver {
```

Comment:
At this point, what is the benefit of having a driver type for _each_ driver? We could potentially take this a step further: everything is a `driver` with a driver description and generic set of ops.